core/shared/src/main/scala/zio/Runtime.scala

```diff
@@ -36,6 +37,31 @@ trait Runtime[+R] {
    */
   val platform: Platform
 
+  /**
+   * Add generic signal handler
+   *
+   * @param signal to listen for
+   * @param handler to run on receiving signal
+   */
+  def addSignalHandler(signal: String)(handler: URIO[R, Unit]): Unit =
+    Platform.addSignalHandler(signal)(() => unsafeRun(handler))
+
+  /**
+   * Use SIGUSR2 signal to interrupt runtime
+   *
+   * @param handler to run when SIGUSR2 is received
+   */
+  def addInterruptHandler(handler: URIO[R, Unit]): Unit =
+    addSignalHandler("USR2")(handler)
+
+  /**
+   * Print Fiber Dump of all fibers when interrupt signal received
+   *
+   * @param ev is the console environment
+   */
+  def printFiberDumpOnInterrupt(implicit ev: R <:< Console): Unit =
```

Comment:
We may want to just use `println` here instead of `console.putStrLn`, which would allow us to simplify this signature and make this always available. We're already doing this at the edge of our program where we are unsafely running this and since it is for debugging purposes I think arguably we would want to print to the standard output even if there is a different console implementation (e.g. a test console). Users who want to do something fancier can always fall back on the more general combinators we created.